src/api/utils/applyDefaultInterceptors.js

```diff
@@ -0,0 +1,45 @@
+import { sessionService } from 'redux-react-native-session';
+import saveSessionHeaders from './saveSessionHeaders';
+
+const ACCESS_TOKEN = 'access-token';
+
+const UNAUTHORIZED = 401;
+
+const defaultRequestInterceptors = [
```

Comment:
I don't quite get this default interceptors concept, why should I create Interceptors outside this module? seems to me that all the Interceptors will be defined inside this or imported here; maybe I'm missing something.

Reply:
I just put it here as a helper, since they are the interceptors that come "by default" on the base to handle session to have some way to set them up easily.

The `applyInterceptors` idea comes from inverting the dependency to keep things decoupled and avoid cases of import loops.

But you are free to call `apiService.request/responseInterceptors.use()` anywhere you want. Usually, you would want it to run before any request (and just once) to get everything set up.

Reply:
Can you think of a case where you want to define an Interceptor after the app starts running? 
What I was thinking is that it could be that in most cases I would want to define Interceptors at the same moment you are defining the default Interceptors; so instead of applyDefaultInterceptors this could be converted to applyInitializingInterceptors or something like that.

Reply:
Analytics maybe? Like, only set up an interceptor if you are logged in.

Reply:
Yes, that have sense hihi.